hail/python/hail/vds/impex.py

```diff
@@ -0,0 +1,335 @@
+    """
+    ref, var = dataset.reference_data, dataset.variant_data
+
+    # fix up ref data
+    if 'END' in ref.entry:
+        ref = ref.drop('END')
+
+    if 'gvcf_info' in var.entry and isinstance(var.gvcf_info.dtype, tstruct):
+        warning(
+            'vds.export_vcf: dropping `gvcf_info` as exporting entry field of type `struct<...>` '
+            'to VCF is not supported. To preserve this information, flatten the struct onto the '
+            'entries.'
+        )
+        var = var.drop('gvcf_info')
+
+    if 'LGT' in var.entry:
+        if 'GT' not in var.entry:
+            var = var.annotate_entries(GT=lgt_to_gt(var.LGT, var.LA))
+        var = var.drop('LGT')
+    if 'LPGT' in var.entry:
+        if 'PGT' not in var.entry:
+            var = var.annotate_entries(PGT=lgt_to_gt(var.LPGT, var.LA))
+        var = var.drop('LPGT')
+
+    if 'LA' in var.entry:
+        var = var.transmute_entries(LAA=var.LA[1:])
+
+    # TODO(chrisvittal):
+    #   In a future enhancement, add appropriate metadata for all 'local' fields. We do not do this
+    #   at this time (Oct. 2024), as htsjdk, which we use for VCF header parsing, does not support
+    #   the vcf 4.5 changes necessary for completely correct metadata.
+
+    extra_header = (
+        '##SVCR="This is an VCF that implements hail\'s Scalable Variant Call '
+        'Representation. See https://doi.org/10.1101/2024.01.09.574205 for more '
+        'information."\n'
+    )
+    if VariantDataset.ref_block_max_length_field in ref.globals:
+        rbml = hl.eval(ref[VariantDataset.ref_block_max_length_field])
+        extra_header += f'##ref_block_max_length={rbml}\n'
+
+    fs = hl.current_backend().fs
+    if append_to_header:
+        with fs.open(append_to_header, 'r') as header_file:
+            extra_header += header_file.read()
+
+    new_header_file_path = hl.utils.new_temp_file('vds-append-to-header', 'txt')
+    with fs.open(new_header_file_path, 'w') as new_header_file:
+        new_header_file.write(extra_header)
+
+    vcf = to_merged_sparse_mt(VariantDataset(reference_data=ref, variant_data=var))
+    hl.export_vcf(vcf, output, append_to_header=new_header_file_path, parallel=parallel, metadata=metadata, tabix=tabix)
+
+
+@typecheck(
+    path=oneof(str, sequenceof(str)),
+    is_split=bool,
+    ref_block_fields=sequenceof(str),
+    infer_ref_block_fields=bool,
+    force=bool,
+    force_bgz=bool,
+    header_file=nullable(str),
+    min_partitions=nullable(int),
+    call_fields=oneof(str, sequenceof(str)),
+    reference_genome=nullable(reference_genome_type),
+    contig_recoding=nullable(dictof(str, str)),
+    array_elements_required=bool,
+    skip_invalid_loci=bool,
+    entry_float_type=enumeration(tfloat32, tfloat64),
+    filter=nullable(str),
+    find_replace=nullable(sized_tupleof(str, str)),
+    n_partitions=nullable(int),
+    block_size=nullable(int),
+    _create_row_uids=bool,
+    _create_col_uids=bool,
+)
+def import_vcf(
```

Comment:
would it make sense to call this `import_svcr` to make it distinct from the extant `import_vcf`?